challenges/challenges.py

```diff
@@ -166,69 +141,41 @@ async def battle_check(self):
-                            if group_plant:
-                                if brawler_name in ("SPIKE", "ROSA", "SPROUT"):
-                                    if win:
-                                        progress += 1
-                                        if brawler_name in wins:
-                                            wins[brawler_name] += 1
-                                        else:
-                                            wins[brawler_name] = 1
-                                    else:
-                                        if brawler_name in loses:
-                                            loses[brawler_name] += 1
-                                        else:
-                                            loses[brawler_name] = 1
+
+                            streak = await self.config.members(user).streak()
+                            if win and player['brawler']['trophies'] >= 500 and battle['battle']['mode'] in ('brawlBall', 'gemGrab', 'bounty', 'siege', 'hotZone'):
+                                streak += 1
+                            elif win and (player['brawler']['trophies'] < 500 or battle['battle']['mode'] not in ('brawlBall', 'gemGrab', 'bounty', 'siege', 'hotZone')):
+                                streak = streak
                             else:
-                                if brawler_name in ("MORTIS", "FRANK", "EMZ"):
-                                    if win:
-                                        progress += 1
-                                        if brawler_name in wins:
-                                            wins[brawler_name] += 1
-                                        else:
-                                            wins[brawler_name] = 1
-                                    else:
-                                        if brawler_name in loses:
-                                            loses[brawler_name] += 1
-                                        else:
-                                            loses[brawler_name] = 1
+                                streak = 0
                         except Exception as e:
                             await error_ch.send(f"{m}\n```py\n{e}```")
                             await error_ch.send(f"{m}\n```py\n{battle}```")
                             continue
-                    
-                    await self.config.member(user).progress.set(members[m]['progress'] + progress)
-                    await self.config.member(user).set_raw('wins', value=wins)
-                    await self.config.member(user).set_raw('loses', value=loses)
+
+                    if streak >= 5:
+                        streak = 0
+                        entries = await self.config.member(user).entries()
+                        entries = entries + 1
+                        await self.config.member(user).entries.set(entries)
+                    if entries >= 20:
+                        entries = 20
```

Comment:
20 entires limit. >= just in case the bot registers several battles at once and shit gets flying.